modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/BinaryUtils.cs

```diff
@@ -1170,6 +1170,38 @@ public static T[] ReadArray<T>(BinaryReader ctx, bool typed)
             return vals;
         }
 
+        /// <summary>
+        /// Read string array.
+        /// </summary>
+        /// <param name="stream">Stream</param>
+        /// <returns>String array.</returns>
+        public static string[] ReadStringArray(IBinaryStream stream)
+        {
+            var len = stream.ReadInt();
+            var res = new string[len];
+
+            for (var i = 0; i < len; i++)
+                res[i] = stream.ReadByte() == HdrNull ? null : ReadString(stream);
```

Comment:
Should not we at least assert here that header is `HdrString`, if not `HdrNull`?

Reply:
Good point, added header checks to this and related methods.